pkg/model/rate.go

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright The Dragonfly Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package model
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+// Rate wraps int64. It is used to parse the custom rate format
+// from YAML and JSON.
+// This type should not propagate beyond the scope of input/output processing.
+type Rate int64
+
+const (
+	B  Rate = 1
+	KB      = 1024 * B
+	MB      = 1024 * KB
+	GB      = 1024 * MB
+)
+
+// Set implements pflag/flag.Value
+func (d *Rate) Set(s string) error {
+	var err error
+	*d, err = ParseRate(s)
+	return err
+}
+
+// Type implements pflag.Value
+func (d *Rate) Type() string {
+	return "rate"
+}
+
+var rateRE = regexp.MustCompile("^([0-9]+)([mMKkBGg])$")
+
+// ParseRate parses a string into a int64.
+func ParseRate(rateStr string) (Rate, error) {
```

Comment:
I am wondering that shall we support pure digital and take `rate.B` as the unit? If not, there may be some compatibility issues.

Reply:
Why adding `B` will cause compatibility issue? I knew that we just support K and M, so IMO adding B is OK?